### PR TITLE
[BACKLOG-21243] MQTT Consumer - Implement Secure Protocol Support

### PR DESCRIPTION
**Pull request by mkambol**
Preparatory change.  Builder pattern for MQTTStreamSource.

https://jira.pentaho.com/browse/BACKLOG-21243

### CODE REVIEW

**Comment by mkambol** [comment]
@pentaho/ackbar

File: kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTStreamSource.java [hunk 1]
```diff
@@ -73,21 +85,32 @@
...
     this.broker = broker;
     this.topics = topics;
     this.qos = qualityOfService;
+    this.isSecure = isSecure;
+    this.username = username;
+    this.password = password;
+    this.sslConfig = sslConfig;
   }
 
+  //  public MQTTStreamSource( String broker, List<String> topics, int qualityOfService,
```
Comment (DFieldFL) [inline kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTStreamSource.java]: should we remove this?

Comment (mkambol) [reply kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTStreamSource.java]: Absolutely- I noticed that after pushing.  I'll clean it out with the next commit.

Comment (DFieldFL) [reply kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTStreamSource.java]: 👍

**Pull request merged**